pkg/security/rules/autosuppression/autosuppression.go

```diff
@@ -40,60 +49,66 @@ type Opts struct {
 
 // StatsTags holds tags for auto suppression stats
 type StatsTags struct {
-	RuleID   string
-	TreeType string
+	RuleID          string
+	SuppressionType string
 }
 
 // AutoSuppression is a struct that encapsulates the auto suppression logic
 type AutoSuppression struct {
-	once             sync.Once
-	opts             Opts
-	statsLock        sync.RWMutex
-	stats            map[StatsTags]*atomic.Int64
-	enabledTreeTypes []string
+	once      sync.Once
+	opts      Opts
+	statsLock sync.RWMutex
+	stats     map[StatsTags]*atomic.Int64
 }
 
 // Init initializes the auto suppression with the given options
 func (as *AutoSuppression) Init(opts Opts) {
 	as.once.Do(func() {
 		as.opts = opts
 		as.stats = make(map[StatsTags]*atomic.Int64)
-		if opts.SecurityProfileAutoSuppressionEnabled {
-			as.enabledTreeTypes = append(as.enabledTreeTypes, securityProfileTreeType)
-		}
-		if opts.ActivityDumpAutoSuppressionEnabled {
-			as.enabledTreeTypes = append(as.enabledTreeTypes, activityDumpTreeType)
-		}
 	})
 }
 
 // Suppresses returns true if the event should be suppressed for the given rule, false otherwise. It also counts statistics depending on this result
 func (as *AutoSuppression) Suppresses(rule *rules.Rule, event *model.Event) bool {
-	if as.opts.SecurityProfileAutoSuppressionEnabled &&
-		event.IsInProfile() &&
-		slices.Contains(as.opts.EventTypes, event.GetEventType()) &&
-		isAllowAutosuppressionRule(rule) {
-		as.count(rule.ID, securityProfileTreeType)
-		return true
-	} else if as.opts.ActivityDumpAutoSuppressionEnabled &&
-		event.HasActiveActivityDump() &&
-		slices.Contains(as.opts.EventTypes, event.GetEventType()) &&
-		isAllowAutosuppressionRule(rule) {
-		as.count(rule.ID, activityDumpTreeType)
-		return true
+	if isAllowAutosuppressionRule(rule) && event.ContainerContext.ID != "" && slices.Contains(as.opts.EventTypes, event.GetEventType()) {
+		if as.opts.ActivityDumpAutoSuppressionEnabled {
+			if event.HasActiveActivityDump() {
+				as.count(rule.ID, activityDumpSuppressionType)
+				return true
+			} else if isWorkloadDriftOnlyRule(rule) && event.SecurityProfileContext.EventTypeState == model.NoProfile {
+				as.count(rule.ID, noWorkloadDriftSuppressionType)
+				return true
+			}
+		}
+		if as.opts.SecurityProfileAutoSuppressionEnabled {
+			if event.IsInProfile() {
+				as.count(rule.ID, securityProfileSuppressionType)
+				return true
+			}
+		}
 	}
 	return false
 }
 
 // Apply resets the auto suppression stats based on the given ruleset
 func (as *AutoSuppression) Apply(ruleSet *rules.RuleSet) {
+	var enabledSuppressionTypes []string
+	if as.opts.SecurityProfileAutoSuppressionEnabled {
+		enabledSuppressionTypes = append(enabledSuppressionTypes, securityProfileSuppressionType)
+		enabledSuppressionTypes = append(enabledSuppressionTypes, noWorkloadDriftSuppressionType)
+	}
+	if as.opts.ActivityDumpAutoSuppressionEnabled {
+		enabledSuppressionTypes = append(enabledSuppressionTypes, activityDumpSuppressionType)
+	}
```

Comment:
I _**think**_ Yoann wanted to stick the "workload drift suppression" to the "activity dump suppression" no (or at least that's what he wrote in the description of the PR) ?

Should we have a third config parameter ? 🤔

Reply:
Good catch!
At start it was under the "security profile" config but I moved it to the "activity dump" one. And that's me who wrote the desc :smile: 
Will be fixed on upcoming commit